sdk/samples/iot/aziot_realtek_amebaD/aziot_realtek_amebaD.ino

```diff
@@ -0,0 +1,385 @@
+  az_span sas_base64_encoded_signed_signature = AZ_SPAN_FROM_BUFFER(b64enc_hmacsha256_signature);
+  iot_sample_generate_sas_base64_encoded_signed_signature(
+    //AZ_SPAN_FROM_BUFFER(device_key),
+    device_key,
+    sas_signature,
+    sas_base64_encoded_signed_signature,
+    &sas_base64_encoded_signed_signature);
+  
+  // Get the resulting MQTT password, passing the base64 encoded, HMAC signed bytes.
+  size_t mqtt_password_length;
+  rc = az_iot_hub_client_sas_get_password(
+    &hub_client,
+    sas_duration,
+    sas_base64_encoded_signed_signature,
+    AZ_SPAN_EMPTY,
+    sas_token,
+    sizeof(sas_token),
+    &sas_token_length);
+  if (az_result_failed(rc))
+  {
+    Serial.print("Could not get the password: az_result return code ");
+    Serial.println(rc);
+  }
+}
+
+static int connectToAzureIoTHub()
+{
+  size_t client_id_length;
+  char mqtt_client_id[128];
+  if (az_result_failed(az_iot_hub_client_get_client_id(
+          &hub_client, mqtt_client_id, sizeof(mqtt_client_id) - 1, &client_id_length)))
+  {
+    Serial.println("Failed getting client id");
+    return 1;
+  }
+
+  mqtt_client_id[client_id_length] = '\0';
+
+  char mqtt_username[128];
+  // Get the MQTT user name used to connect to IoT Hub
+  if (az_result_failed(az_iot_hub_client_get_user_name(
+          &hub_client, mqtt_username, sizeofarray(mqtt_username), NULL)))
+  {
+    printf("Failed to get MQTT clientId, return code\n");
+    return 1;
+  }
+
+  Serial.print("Client ID: ");
+  Serial.println(mqtt_client_id);
+
+  Serial.print("Username: ");
+  Serial.println(mqtt_username);
+
+  while (!mqtt_client.connected())
+  {
+    Serial.print("MQTT connecting ... ");
+
+    if (mqtt_client.connect(mqtt_client_id, mqtt_username, sas_token))
+    {
+      Serial.println("connected.");
+    }
+    else
+    {
+      Serial.print("failed, status code =");
+      Serial.print(mqtt_client.state());
+      Serial.println(". Try again in 5 seconds.");
+      // Wait 5 seconds before retrying
+      delay(5000);
+    }
+  }
+
+  mqtt_client.subscribe(AZ_IOT_HUB_CLIENT_C2D_SUBSCRIBE_TOPIC);
+
+  return 0;
+}
+
+void establishConnection() 
+{
+  connectToWiFi();
+  initializeClients();
+
+  generate_sas_key();
+
+  connectToAzureIoTHub();
+
+  digitalWrite(LED_PIN, LOW);
+}
+
+void setup()
+{
+  pinMode(LED_PIN, OUTPUT);
+  digitalWrite(LED_PIN, HIGH);
+  createNullTerminatedRootCert();
+  establishConnection();
+}
+
+static char* getTelemetryPayload()
+{
+  az_span temp_span = az_span_create(telemetry_payload, sizeof(telemetry_payload));
+  temp_span = az_span_copy(temp_span, AZ_SPAN_FROM_STR("{ \"deviceId\": \"" IOT_CONFIG_DEVICE_ID "\", \"msgCount\": "));
+  (void)az_span_u32toa(temp_span, telemetry_send_count++, &temp_span);  
+  temp_span = az_span_copy(temp_span, AZ_SPAN_FROM_STR(" }"));
+  temp_span = az_span_copy_u8(temp_span, '\0');
+
+  return (char*)telemetry_payload;
+}
+
+static void sendTelemetry()
```

Comment:
```suggestion
static void send_telemetry()
```

Reply:
That's because some functions were copied from Realtek repo. Their style is different. But ok, I'll take this fix.